source/includes/language-compatibility-table-java.rst

```diff
@@ -0,0 +1,90 @@
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :class: compatibility-large
+
+   * - Java Driver Version
+     - Java 5
+     - Java 6
+     - Java 7
+     - Java 8
+     - Java 11 [*]_
+
+   * - Version 3.12
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.11
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.10
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.9
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.8
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.4
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.3
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.2
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.1
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Version 3.0
+     -
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+   * - Versions >= 2.7
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+.. [*] Java versions 8 and above are all supported thanks to the JVM backwards compatibility promise. Only LTS versions will be explicitly listed in future.
```

Comment:
There is no JVM backward compatibility promise per se.  We are starting to see deprecated APIs being removed in newer JVM versions.